### ISSUE TITLE
do connect error: lookup localhost: no such host

### ISSUE THREAD

**Issue by haliliceylan** [kubenetworks/kubevpn#181]
➜  ~ kubevpn connect
start to connect
got cidr from cache
get cidr successfully
update ref count successfully
traffic manager already exist, reuse it
do connect error: lookup localhost: no such host
prepare to exit, cleaning up
update ref count successfully
no proxy resources found
clean up successfully
Error: rpc error: code = Unknown desc = lookup localhost: no such host

**Comment by haliliceylan** [comment]
➜  ~ kubevpn reset  
Error: http: Server closed
Error: http: Server closed
no proxy resources found
leave proxy resources success
cleanup k8s resource
done%

**Comment by wencaiwulue** [comment]
yes, maybe it's a bug, i will try to find the reason. keep waitting for my good news ~

**Comment by wencaiwulue** [comment]
@haliliceylan it should resolved in this commit https://github.com/kubenetworks/kubevpn/commit/0d64dc7b10d5bb18793df3dcbd12fd731bf7093e. already released new version, you can have a try ~

**Comment by haliliceylan** [comment]
Yeah it just opened right now, let me check.

**Comment by haliliceylan** [comment]
sorry but still I can experience the same issue @wencaiwulue 

```
➜  ~ cat /etc/hosts          
##

127.0.0.1	localhost%                                                      
➜  ~ kubevpn disconnect --all
disconnect successfully%                                                        
➜  ~ kubevpn connect         
start to connect
got cidr from cache
get cidr successfully
update ref count successfully
traffic manager already exist, reuse it
do connect error: lookup localhost: no such host
prepare to exit, cleaning up
update ref count successfully
no proxy resources found
clean up successfully
Error: rpc error: code = Unknown desc = lookup localhost: no such host
➜  ~ cat /etc/hosts          
##
➜ ~ kubevpn version
KubeVPN: CLI
    Version: v2.2.3
    DaemonVersion: v2.2.3
    Image: docker.io/naison/kubevpn:v2.2.3
    Branch: HEAD
    Git commit: 0d64dc7b10d5bb18793df3dcbd12fd731bf7093e
    Built time: 2024-03-03 11:48:51
    Built OS/Arch: linux/amd64
    Built Go version: go1.20.14
```

**Comment by haliliceylan** [comment]
When you run the command `kubevpn disconnect --all` consecutively, this problem occurs.

<img width="354" alt="Screenshot 2024-03-03 at 16 36 12" src="https://github.com/kubenetworks/kubevpn/assets/33789758/b6f8985b-7a60-4882-bf10-082dcbc193b3">

**Comment by wencaiwulue** [comment]
> sorry but still I can experience the same issue @wencaiwulue
> 
> ```
> ➜  ~ cat /etc/hosts          
> ##
> 
> 127.0.0.1	localhost%                                                      
> ➜  ~ kubevpn disconnect --all
> disconnect successfully%                                                        
> ➜  ~ kubevpn connect         
> start to connect
> got cidr from cache
> get cidr successfully
> update ref count successfully
> traffic manager already exist, reuse it
> do connect error: lookup localhost: no such host
> prepare to exit, cleaning up
> update ref count successfully
> no proxy resources found
> clean up successfully
> Error: rpc error: code = Unknown desc = lookup localhost: no such host
> ➜  ~ cat /etc/hosts          
> ##
> ➜ ~ kubevpn version
> KubeVPN: CLI
>     Version: v2.2.3
>     DaemonVersion: v2.2.3
>     Image: docker.io/naison/kubevpn:v2.2.3
>     Branch: HEAD
>     Git commit: 0d64dc7b10d5bb18793df3dcbd12fd731bf7093e
>     Built time: 2024-03-03 11:48:51
>     Built OS/Arch: linux/amd64
>     Built Go version: go1.20.14
> ```
> 
> As a temporary solution, I'm manually adding the localhost entry to etc hosts. However, deleting localhost from /etc/hosts is really bad, as it makes localhost unresolvable in the system.

yes, delete localhost is bad, but i tested on my macbook and linux server, all works fine, i am comfusing

**Comment by wencaiwulue** [comment]
> When you run the command `kubevpn disconnect --all` consecutively, this problem occurs.
> 
> <img alt="Screenshot 2024-03-03 at 16 36 12" width="354" src="https://private-user-images.githubusercontent.com/33789758/309561330-b6f8985b-7a60-4882-bf10-082dcbc193b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0ODA2MDcsIm5iZiI6MTcwOTQ4MDMwNywicGF0aCI6Ii8zMzc4OTc1OC8zMDk1NjEzMzAtYjZmODk4NWItN2E2MC00ODgyLWJmMTAtMDgyZGNiYzE5M2IzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAzVDE1MzgyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1MTY5NzJkOTBjYmEwM2ViM2Q4NTIxOGEzNDJjZGFjMTFlZTI3MzQ4ZmRkMGIyZDI4YjQ4MjdiYWQ5MWMwOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xElbMRMGb3Tf38aMblZOZyrqtK6YZpa9XFT1TYSOs2I">

you are right, fix it right now

**Comment by haliliceylan** [comment]
I completely understand the error now,

For some reason, there are two "#" signs at the beginning of the /etc/hosts file. When these signs are present, and I run the disconnect command consecutively, localhost gets deleted. However, if the /etc/hosts file is completely empty and there is a localhost entry in the first line, it seems like the error does not occur. Similarly, when I leave the first line of /etc/hosts empty and write the localhost entry in the second line, it gets deleted again. @wencaiwulue

**Comment by wencaiwulue** [comment]
> I completely understand the error now,
> 
> For some reason, there are two "#" signs at the beginning of the /etc/hosts file. When these signs are present, and I run the disconnect command consecutively, localhost gets deleted. However, if the /etc/hosts file is completely empty and there is a localhost entry in the first line, it seems like the error does not occur. Similarly, when I leave the first line of /etc/hosts empty and write the localhost entry in the second line, it gets deleted again. @wencaiwulue

yes, i found the bug, only need 5 min to fix it

**Comment by wencaiwulue** [comment]
<img width="1207" alt="image" src="https://github.com/kubenetworks/kubevpn/assets/33818115/15ec342c-a6eb-42c4-955c-89fe9d1de857">


> I completely understand the error now,
> 
> For some reason, there are two "#" signs at the beginning of the /etc/hosts file. When these signs are present, and I run the disconnect command consecutively, localhost gets deleted. However, if the /etc/hosts file is completely empty and there is a localhost entry in the first line, it seems like the error does not occur. Similarly, when I leave the first line of /etc/hosts empty and write the localhost entry in the second line, it gets deleted again. @wencaiwulue

**Comment by wencaiwulue** [comment]
I am redo-releaseing version v2.2.3, this needs about 20 min

**Comment by wencaiwulue** [comment]
@haliliceylan already redo-released version v2.2.3, you can use command `kubevpn upgrade` to upgrade it, this bug is resolved finally, thanks very much ~

**Comment by haliliceylan** [comment]
Thank you very much !
